modules/tensor_mechanics/src/materials/ADHillPlasticityStressUpdate.C

```diff
@@ -131,6 +101,7 @@ ADHillPlasticityStressUpdate::computeDeltaDerivatives(const ADReal & delta_gamma
 {
   omega_gamma = 0.0;
   sy_gamma = 0.0;
+  computeHillTensorEigenDecomposition(_hill_tensor[_qp]);
```

Comment:
Can we compute the Eigendecomposition once per time step?
Doing that will involve some approximation once we account for the kinematics of the rotation increment. But, at the same time, omitting some inner loop decompositions should not affect convergence and may help us with computational efficiency. As the tests are set up, probably the eigendecomposition can be moved to an outer layer.